### PR TITLE
Account for non deterministic ipv6 address

### PR DESCRIPTION
**Pull request by vandabarata**
## What does this PR change?
Account for non deterministic IPv6 address being shown

## GUI diff
No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- Cucumber tests were modified

- [x] **DONE**

## Ports
- [4.1](https://github.com/SUSE/spacewalk/pull/17056)
- [4.2](https://github.com/SUSE/spacewalk/pull/17055)

- [ ] **DONE**

## Changelogs
- [x] No changelog needed

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by srbarrios — APPROVED** [review]

File: testsuite/features/step_definitions/common_steps.rb [hunk 1]
```diff
@@ -53,14 +53,16 @@
...
-  ipv6_line = lines.grep(/inet6 2620:[:0-9a-f]*\/64 scope global dynamic mngtmpaddr/).first if ipv6_line.nil?
-  ipv6_line = lines.grep(/inet6 fe80:[:0-9a-f]*\/64 scope link/).first if ipv6_line.nil?
-  next if ipv6_line.nil?
-  ipv6_address = ipv6_line.scan(/2620:[:0-9a-f]*|fe80:[:0-9a-f]*/).first
+  # selects only lines with IPv6 addresses and proceeds to form an array with only those addresses
+  ipv6_addresses_list = lines.grep(/2620:[:0-9a-f]*|fe80:[:0-9a-f]*/)
+  ipv6_addresses_list.map! { |ip_line| ip_line.slice(/2620:[:0-9a-f]*|fe80:[:0-9a-f]*/) }
+
+  # confirms that the IPv6 address shown on the page is part of that list and, therefore, valid
+  ipv6_address = find(:xpath, '//*[@id="spacewalk-content"]/div[4]/div[1]/div[1]/div[2]/table/tbody/tr[3]/td[2]').text
```
Comment (srbarrios) [inline testsuite/features/step_definitions/common_steps.rb:63]: I personally don't like too much that kind of XPATHs based on HTML structure, I rather prefer queries using IDs, Names, CSS content, whatever. BUT, let's merge this, and maybe we can improve it in the future.

**Review by nodeg — APPROVED** [review]
LGTM.

**Review by Bischoff — APPROVED** [review]
Same remark as Oscar. If there is an id available it would be much better.

File: testsuite/features/step_definitions/common_steps.rb (hunk 1 above)
Comment (nodeg) [reply testsuite/features/step_definitions/common_steps.rb:63]: There are definitely some IDs/classes we can use here.
image

Comment (Bischoff) [reply testsuite/features/step_definitions/common_steps.rb:63]: or just do like this:
```
<tr>
            <td>IPv6 Address:</td>
            <td>
                fe80::a8b2:92ff:fe42:97
            </td>
</tr>

td[text()='IPv6 Address']/next-sibling()
```

**Review by Bischoff — CHANGES_REQUESTED** [review]
simplify xpath with `//"//td[text()='IPv6 Address:']/following-sibling::td[1]/text()` (please test in cucumber first, as you might need to trim the linefeeds)

**Pull request merged**